### DISCUSSION TITLE
Mapping to send foreground process to background and vice versa

### DISCUSSION THREAD

**Discussion by ariel-frischer** [kovidgoyal/kitty#6451] [Q&A]
I want one mapping to simulate calling `ctrl+z` (send to background), then running the `fg` command, to bring it back to the foreground. My use case is for neovim, and I don't like the built in neovim terminal stuff.
Best I have so far is:
```
map super+l signal_child SIGTSTP
```
This seems to kinda work for hiding the vim process, but doesn't redraw the screen, it also doesn't alternate clearly.

**Comment by kovidgoyal** [comment]
If you want one mapping to do multiple things depending on state,
you need to write a kitten for it. See the many third party kittens for
inspiration. You can have your kitten check if the foreground process is
vim and if so send it the stop signal and do whatever else like clear
screen, switch to main screen etc and if not do whatever is neeeded to
restore it.
